### PR TITLE
EVEREST-488 Added version command

### PR DESCRIPTION
**Pull request by gen1us2k**
**CHANGE DESCRIPTION**
---
**Problem:**
EVEREST-488

**Solution:**
*Short explanation of the solution we are providing with this PR.*

Implementation of version command
Added support of the command for the release 
Added support of automatic setting the version of the catalog 

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?

**Tests**
- [x] Is an Integration test/test case added for the new feature/change?
- [x] Are unit tests added where appropriate?

### CODE REVIEW

File: commands/version.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	"go.uber.org/zap"
+
+	"github.com/percona/percona-everest-cli/pkg/version"
+)
+
+func newVersionCmd(_ *zap.SugaredLogger) *cobra.Command {
+	return &cobra.Command{
+		Use: "version",
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println(version.FullVersionInfo()) //nolint:forbidigo
```
Comment (michal-kralik) [inline commands/version.go:16]: Can you please add support for the global `--json` flag?
See https://github.com/percona/percona-everest-cli/blob/main/pkg/output/command.go#L33

**Review by oksana-grishchenko — APPROVED** [review]

File: commands/version.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+				return
+			}
+			if !outputJSON {
+				fmt.Println(version.FullVersionInfo()) //nolint:forbidigo
+			}
+			version, err := version.FullVersionJSON()
+			if err != nil {
+				l.Errorf("could not print JSON. Error: %s", err)
+			}
+			fmt.Println(version) //nolint:forbidigo
```
Comment (michal-kralik) [inline commands/version.go:28]: Here's an example how the output module can be used to format result as a json
https://github.com/percona/percona-everest-cli/blob/18a80a6efa4fd10a1c3bfcebd15e91d94c45799d/commands/list/versions.go#L60

Comment (michal-kralik) [reply commands/version.go:28]: maybe this one is easier for illustration
https://github.com/percona/percona-everest-cli/blob/18a80a6efa4fd10a1c3bfcebd15e91d94c45799d/commands/list/database_engines.go#L60

Comment (gen1us2k) [reply commands/version.go:28]: the output of the command is just a set of strings that cannot be marshalled into json properly, hence the existing implementation.

Comment (michal-kralik) [reply commands/version.go:28]: I don't understand. You already return a json in https://github.com/percona/percona-everest-cli/pull/174/files#diff-73b8ec8a05213857da6e9598e34c5d7a18a6e5349250908ae6710578236977f2R65
Am I missing anything since you're saying it cannot be marshalled into json properly?

The codebase has a slightly different approach to running commands. 
In `commands/` we initialize a structure and call `Run()` on it. All logic for commands then resides in `pkg/<command_name>`. This makes the commands loosely coupled with the rest of the codebase and allows for a better unit testing.

The other commands, which print json output, return a structure which then can be marshalled into a json response or string response.
I think it would make sense to keep the approach consistent across our codebase and make the `version` command follow the same approach.

Comment (gen1us2k) [reply commands/version.go:28]: I don't believe that we need to keep the codebase consistent for version command that uses string variables and I added `json` flag just to keep global vars consistent. 

On the other hand, this looks a bit redundant. The version command does not use any structs and uses a map to have JSON output.

I'd rather drop JSON support for this command to keep everything easy and right now we do not have a requirement to support JSON output for the CLI. Hence, I'd rather drop JSON logic and keep the command as simple as we can.

@michal-kralik Why do we need to have JSON support for version command? What are the usecases for this?

Comment (michal-kralik) [reply commands/version.go:28]: How would you suggest `everestctl` is used in automation? Do we expect users to parse stdout with regular expressions and we keep the log messages and format consistent across versions?

I would rather prefer `everestctl` to return a consistent json output.
You can always keep `map[string]string` as the result and implement a `String()` method which returns a non-json output.

Comment (gen1us2k) [reply commands/version.go:28]: I'm not sure that `everestctl version` will be part of automation anyway

Comment (recharte) [reply commands/version.go]: > I'm not sure that everestctl version will be part of any automation.
It can be hard to picture it at this stage given the early stages of the project and because we haven't had a version command until now but I do see this as a useful feature. If it's not a big effort (I don't think it is) I would add this json output because it can definitely be useful.

Comment (gen1us2k) [reply commands/version.go:28]: Added but without using output package. It works and I'm not sure that adding this information as struct/map/custom type makes sense

File: cli-tests/tests/flow/pg-operator.spec.ts [hunk 3]
```diff
@@ -73,9 +73,6 @@ test.describe('Everest CLI install operators', async () => {
       await restartedOperator.assertSuccess();
 
       expect(operator.getStdOutLines()[0]).not.toEqual(restartedOperator.getStdOutLines()[0]);
-      const out = await cli.everestExecSkipWizard(
-        `install operators --operator.mongodb=false --operator.postgresql=true --operator.xtradb-cluster=true --backup.enable=0 --monitoring.enable=0 --name=${clusterName}`,
-      );
```
Comment (recharte) [inline cli-tests/tests/flow/pg-operator.spec.ts:78]: Why was this removed?

File: commands/version.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	return &cobra.Command{
+		Use: "version",
+		Run: func(cmd *cobra.Command, args []string) {
+			outputJSON, err := cmd.Flags().GetBool("json")
+			if err != nil {
+				l.Errorf("could not parse json global flag. Error: %s", err)
+				return
+			}
+			if !outputJSON {
+				fmt.Println(version.FullVersionInfo()) //nolint:forbidigo
```
Comment (recharte) [inline commands/version.go:22]: Shouldn't we return here? Without the return won't we have the JSON as well?

File: pkg/version/version.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	goversion "github.com/hashicorp/go-version"
+)
+
+const (
+	devCatalogImage     = "docker.io/percona/everest-catalog:latest"
```
Comment (recharte) [inline pkg/version/version.go:29]: Not solely related with this PR but we should review this, `latest` should be used as an alias for the latest stable version and not dev builds.

File: cli-tests/tests/flow/pg-operator.spec.ts (hunk 3 above)
Comment (gen1us2k) [reply cli-tests/tests/flow/pg-operator.spec.ts:78]: Because it's redundant

Comment (gen1us2k) [reply cli-tests/tests/flow/pg-operator.spec.ts:78]: and this was merge artifact

**Review by recharte — APPROVED** [review]
